### DISCUSSION TITLE
Inserted dates wrong every 2048th row

### DISCUSSION THREAD

**Discussion by slvr611** [duckdb/duckdb#10101] [Q&A] [answered]
I am programming in python using DuckDB to interface with a sql database and DBeaver to view the database.  I'm getting a lot of input data that is entered into my database in batches of around 1.8k rows at a time.  Every 2048th row, starting at row 2049 is having a problem where when viewing the row in a full table view it displays everything correctly, but when querying for the row with that specific ID it has the wrong date and sometimes another integer on the row is wrong as well, sometimes negative when it should only ever be positive, although this is not consistent.  To me that indicates something overflow related.  This wrong date persists into queries from my main python script using DuckDB, but the really strange thing to me is that it displays the correct date in the full table view, but when queried for it displays the incorrect date.

My inclination is that the problem is related to the underlying database, as it's every 2048th row regardless of the size the batches are inserted in, and I suspect the problem is DuckDB related because it occurs directly on insertion into the database, although it could be some property of the SQL database itself that is messing with things rather than DuckDB.

Any input and advice is greatly appreciated, and I will gladly share additional info as needed, although some of it may need to be via DM.

**Comment by Tishj** [comment, marked as answer]
That sounds like a bug, and we would appreciate it if you could make an issue instead of a discussion, with a reproducable example attached 👍

**Reply by slvr611** [reply]
Thanks, I'll do that shortly

**Comment by Giorgi** [comment]
I wonder if this is somehow related to [https://github.com/Giorgi/DuckDB.NET/issues/160](https://github.com/Giorgi/DuckDB.NET/issues/160)

**Reply by slvr611** [reply]
Yes, this does look the same!  In that issue it is listed as having a 1 in several thousand chance of happening, and that it always appears on the same row.  Both those point to being very similar to this problem's ocurrance of every 2048th row

It doesn't look like that other problem was ever fully solved, or is there something I'm missing?

**Reply by Giorgi** [reply]
No, I haven't looked into it. But even if I had I would not be able to diagnose it because I only built a .Net wrapper around DuckDB.

It can probably serve as a repro.
